website/blog/2024-03-21-using-your-private-git-repository-context-for-completion/index.md

```diff
@@ -0,0 +1,113 @@
+
+## Building the Index
+
+:::tip
+The commands provided in this section are based on a Linux environment and assume the pre-installation of Docker with CUDA drivers. Adjust the commands as necessary if you are running Tabby on a different setup.
+:::
+
+Once the configuration file is set up, proceed with running the `scheduler` to synchronize git repositories and construct the index. In this scenario, utilizing the `tabby-cpu` entrypoint will avoid the requirement for GPU resources.
+
+```bash
+docker run -it --entrypoint /opt/tabby/bin/tabby-cpu -v $HOME/.tabby:/data tabbyml/tabby scheduler --now
+```
+
+The expected output looks like this:
+
+```console
+icy@Icys-Ubuntu:~$ docker run -it --entrypoint /opt/tabby/bin/tabby-cpu -v $HOME/.tabby:/data tabbyml/tabby scheduler --now
+Syncing 1 repositories...
+Cloning into '/data/repositories/my_private_project'...
+remote: Enumerating objects: 51, done.
+remote: Total 51 (delta 0), reused 0 (delta 0), pack-reused 51
+Receiving objects: 100% (51/51), 7.16 KiB | 2.38 MiB/s, done.
+Resolving deltas: 100% (18/18), done.
+Building dataset...
+100%|████████████████████████████████████████| 12/12 [00:00<00:00, 55.56it/s]
+Indexing repositories...
+100%|████████████████████████████████████████| 12/12 [00:00<00:00, 73737.70it/s]
+```
+
+Subsequently, launch the server using the following command:
+
+```bash
+docker run -it --gpus all -p 8080:8080 -v $HOME/.tabby:/data tabbyml/tabby serve --model StarCoder-1B --device cuda
+```
+
+The expected output upon successful initiation of the server should like this:
+
+```console
+icy@Icys-Ubuntu:~$ docker run -it --gpus all -p 8080:8080 -v $HOME/.tabby:/data tabbyml/tabby serve --model StarCoder-1B --device cuda
+2024-03-21T16:16:47.189632Z  INFO tabby::serve: crates/tabby/src/serve.rs:118: Starting server, this might take a few minutes...
+2024-03-21T16:16:47.190764Z  INFO tabby::services::code: crates/tabby/src/services/code.rs:53: Index is ready, enabling server...
+ggml_init_cublas: GGML_CUDA_FORCE_MMQ:   no
+ggml_init_cublas: CUDA_USE_TENSOR_CORES: yes
+ggml_init_cublas: found 1 CUDA devices:
+  Device 0: NVIDIA GeForce RTX 4090, compute capability 8.9, VMM: yes
+2024-03-21T16:16:52.464116Z  INFO tabby::routes: crates/tabby/src/routes/mod.rs:35: Listening at 0.0.0.0:8080
+```
+
+Notably, the line `Index is ready, enabling server...` signifies that the server has been successfully launched with the constructed index.
+
+## Verifying Indexing Results
+
+To confirm that the code completion is effectively utilizing the built index, you can employ the code search feature to validate the indexing process:
+
+1. Access the Swagger UI page: `http://localhost:8080/swagger-ui/#/v1beta/search`.
+2. Click on the `Try it out` button, and input the query parameter `q`, such as `get`.
+3. Click the `Execute` button to trigger the search and see if there are any indexed symbols.
+
```

Comment:
Add a screenshot for the code search experience?

Reply:
Updated in [01e4d76](https://github.com/TabbyML/tabby/pull/1705/commits/01e4d7675bc7af89b1b088f8bddff944df184412)